pkg/questionModel/keyValue.go

```diff
@@ -0,0 +1,247 @@
+			kv.err = errors.New("User has quit before finishing question!")
+			return kv, tea.Quit
+
+		case tea.KeyUp, tea.KeyDown, tea.KeyEnter, tea.KeyShiftTab, tea.KeyTab:
+			msgType := msg.Type
+
+			if msgType == tea.KeyEnter && kv.focusIndex == len(kv.inputs) {
+				if !kv.submitButtonFocused {
+					kv.addTag()
+				} else {
+					return kv, tea.Quit
+				}
+			}
+
+			kv.cycleFocusIndex(msgType)
+
+			if kv.focusIndex != len(kv.inputs) {
+				kv.submitButtonFocused = false
+			}
+
+			cmds := make([]tea.Cmd, len(kv.inputs))
+			for i := 0; i <= len(kv.inputs)-1; i++ {
+				if i == kv.focusIndex {
+					cmds[i] = kv.focusInput(i)
+					continue
+				}
+				kv.inputs[i].Blur()
+				kv.inputs[i].PromptStyle = smallLeftPadding
+				kv.inputs[i].TextStyle = noStyle
+			}
+
+			return kv, tea.Batch(cmds...)
+
+		case tea.KeyRight, tea.KeyLeft:
+			if kv.focusIndex == len(kv.inputs) {
+				kv.submitButtonFocused = !kv.submitButtonFocused
+			}
+
+		case tea.KeyBackspace:
+			kv.deleteTag()
+			if len(kv.tagList.list.Items()) == 0 {
+				kv.focusIndex = 0
+				return kv, kv.focusInput(kv.focusIndex)
+			}
+		}
+
+	case error:
+		kv.err = msg
+		return kv, tea.Quit
+	}
+
+	cmd := kv.updateInputs(msg)
+
+	return kv, cmd
+}
+
+func (kv *keyValue) View() string {
+	var b strings.Builder
+	if kv.question != "" {
+		b.WriteString(kv.question + "\n")
+	}
+
+	if len(kv.tagList.list.Items()) > 0 {
+		b.WriteString("\n" + kv.tagList.View() + "\n")
+	} else {
+		b.WriteRune('\n')
+	}
+
+	for i := range kv.inputs {
+		b.WriteString(kv.inputs[i].View())
+		if i < len(kv.inputs)-1 {
+			b.WriteRune('\n')
+		}
+	}
+
+	addButton := kv.createButton(addButtonText, !kv.submitButtonFocused)
+	submitButton := kv.createButton(submitButtonText, kv.submitButtonFocused)
+	b.WriteString(fmt.Sprintf(smallLeftPadding.Render("\n\n%s  %s\n"), addButton, submitButton))
+
+	return b.String()
+}
+
+func (kv *keyValue) addTag() {
+	if kv.inputs[0].Value() == "" {
+		kv.inputs[0].Placeholder = "Please Enter A Key!"
+	}
+
+	if kv.inputs[1].Value() == "" {
+		kv.inputs[1].Placeholder = "Please Enter A Value!"
+	}
+
+	if kv.inputs[0].Value() != "" && kv.inputs[1].Value() != "" {
+		kv.tags = append(kv.tags, []string{strings.TrimSpace(kv.inputs[0].Value()), strings.TrimSpace(kv.inputs[1].Value())})
+		kv.tagList.InitializeModel(&BubbleTeaData{
+			OptionData:    kv.tags,
+			HeaderStrings: []string{"KEY", "VALUE"},
+		})
+		kv.inputs[0].Placeholder = "Key"
+		kv.inputs[1].Placeholder = "Value"
+		kv.inputs[0].SetValue("")
+		kv.inputs[1].SetValue("")
+	}
+}
+
+func (kv *keyValue) createButton(buttonText string, isFocused bool) string {
+	button := blurred.Copy().Render(fmt.Sprintf("[ %s ]", buttonText))
+	if kv.focusIndex == len(kv.inputs) && isFocused {
+		button = fmt.Sprintf("[ %s ]", focused.Render(buttonText))
+	}
+	return button
+}
+
+func (kv *keyValue) cycleFocusIndex(msgType tea.KeyType) {
+	if msgType == tea.KeyUp || msgType == tea.KeyShiftTab {
+		kv.focusIndex--
+	} else {
```

Comment:
Yes. When the user hits enter after typing something in it will go to the next input, or to the add tag button. Also after using the add tag button the cursor will go back to the Key text box